src/imitation/algorithms/adversarial.py

```diff
@@ -443,16 +435,17 @@ def __init__(
         discrim = discrim_net.DiscrimNetGAIL(
             venv.observation_space, venv.action_space, **discrim_kwargs
         )
-        super().__init__(venv, gen_policy, discrim, expert_data, **kwargs)
+        super().__init__(venv, gen_algo, discrim, expert_data, **kwargs)
 
 
 class AIRL(AdversarialTrainer):
     def __init__(
         self,
         venv: vec_env.VecEnv,
         expert_data: Union[types.Transitions, dataset.Dataset[types.Transitions]],
-        gen_policy: base_class.BaseRLModel,
+        gen_algo: base_class.BaseAlgorithm,
         *,
+        # FIXME(sam): pass in reward net directly, not via _cls and _kwargs
         reward_net_cls: Type[reward_net.RewardNet] = reward_net.BasicShapedRewardNet,
```

Comment:
Don't do this; instead, pass in fully instantiated reward and discriminator networks.

Reply:
Leaving this for current PR. Will make the reward/discriminator interface more uniform in a future commit.